mesonbuild/dependencies/base.py

```diff
@@ -1847,15 +1847,19 @@ def _windows_sanitize_path(path: str) -> str:
         # Ensure that we use USERPROFILE even when inside MSYS, MSYS2, Cygwin, etc.
         if 'USERPROFILE' not in os.environ:
             return path
-        # Ignore executables in the WindowsApps directory which are
-        # zero-sized wrappers that magically open the Windows Store to
-        # install the application.
+        # Order executables in the WindowsApps directory at the end of
+        # the search. These apps might magically open the Windows
+        # Store to install the application, or might redirect to the
+        # installed application.
         appstore_dir = Path(os.environ['USERPROFILE']) / 'AppData' / 'Local' / 'Microsoft' / 'WindowsApps'
         paths = []
+        appstore_paths = []
         for each in path.split(os.pathsep):
             if Path(each) != appstore_dir:
                 paths.append(each)
-        return os.pathsep.join(paths)
+            else:
+                appstore_paths.append(each)
```

Comment:
Why are we adding `appstore_dir` multiple times in the end? Shouldn't we just add it once?